profiler/profiler.go

```diff
@@ -78,12 +78,25 @@ func newProfiler(opts ...Option) (*profiler, error) {
 	if os.Getenv("DD_PROFILING_WAIT_PROFILE") != "" {
 		cfg.addProfileType(expGoroutineWaitProfile)
 	}
-	if cfg.apiKey != "" {
+	// Agentless upload is disabled by default as of v1.30.0, but
+	// WithAgentlessUpload can be used to enable it for testing and debugging.
+	if cfg.agentless {
 		if !isAPIKeyValid(cfg.apiKey) {
-			return nil, errors.New("API key has incorrect format")
+			return nil, errors.New("profiler.WithAgentlessUpload requires a valid API key. Use profiler.WithAPIKey or the DD_API_KEY env variable to set it")
 		}
+		// Always warn people against using this mode for now. All customers should
+		// use agent based uploading at this point.
+		log.Warn("profiler.WithAgentlessUpload is currently for internal usage only and not officially supported. You should not enable it unless somebody at Datadog instructed you to do so.")
```

Comment:
```suggestion
		log.Warn("profiler.WithAgentlessUpload is currently for internal usage only and not officially supported.")
```
IMHO the last part is clear. If someone does tell a user to use it, they will know that they will get this warning and that it is ok.